hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/Schema.java

```diff
@@ -179,8 +179,8 @@ public void readData(ObjectDataInput in) throws IOException {
         for (int i = 0; i < fieldDefinitionsSize; i++) {
             String fieldName = in.readString();
             byte type = in.readByte();
-            FieldType fieldType = FieldType.get(type);
-            FieldDescriptor descriptor = new FieldDescriptor(fieldName, fieldType);
+            FieldKind fieldKind = FieldKind.values()[type];
```

Comment:
Using `values` is pretty inefficient as it allocates an array each time. We should introduce some logic on FieldKind to convert ids to FieldKinds